From 38917442a4a6b5e9cfd03955159f0e69f26e485e Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Wed, 4 Apr 2018 13:02:37 +0300
Subject: [PATCH 129/706] enetc: Update to new BD ring lenght reg (LENR) spec
 as per BGv108+

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit 4de9839e3cfed83494e3fe4329197bada5d4f579)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index da25236a8ff2..0da594cce33e 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -156,8 +156,7 @@ enum enetc_bdr_type {TX, RX};
 #define ENETC_TBICIR0	0xa8
 #define ENETC_TBICIR0_ICEN	BIT(31)
 
-//FIXME , BG 0.7: #define ENETC_RTBMR_RSIZE(n)  __ilog2_u32((n) >> 6)
-#define ENETC_RTBLENR_LEN(n)	(((n) >> 3) << 7)
+#define ENETC_RTBLENR_LEN(n)	((n) & ~0x7)
 #define ENETC_TBMR_EN	BIT(31)
 
 /* Port regs, offset: 1_0000h */
-- 
2.17.1

